HelloWorld_ST_Sensors/user/heartbeat/heartbeat.hpp

```diff
@@ -0,0 +1,223 @@
+    SENSOR_ALGO_COUNTER
+  };
+
+  // Index Byte under the Family Byte: READ_DATA_OUTPUT (0x12)
+  enum FIFO_OUTPUT_INDEX_BYTE
+  {
+    NUM_SAMPLES,
+    READ_DATA
+  };
+
+  // Index Byte under the Family Byte: ENABLE_SENSOR (0x44)
+  enum SENSOR_ENABLE_INDEX_BYTE
+  {
+    ENABLE_MAX30101 = 0x03,
+    ENABLE_ACCELEROMETER
+  };
+
+  // Index Byte under the Family Byte: CHANGE_ALGORITHM_CONFIG (0x50)
+  enum ALGORITHM_CONFIG_INDEX_BYTE
+  {
+    SET_TARG_PERC = 0x00,
+    SET_STEP_SIZE = 0x00,
+    SET_SENSITIVITY = 0x00,
+    SET_AVG_SAMPLES = 0x00,
+    SET_PULSE_OX_COEF = 0x02,
+  };
+
+  // Index Bytes under the Family Byte: READ_ALGORITHM_CONFIG (0x51)
+  enum READ_ALGORITHM_INDEX_BYTE
+  {
+    READ_AGC_PERCENTAGE = 0x00,
+    READ_AGC_STEP_SIZE = 0x00,
+    READ_AGC_SENSITIVITY = 0x00,
+    READ_AGC_NUM_SAMPLES = 0x00,
+    READ_MAX_FAST_COEF = 0x02
+  };
+
+  // Write Bytes under the Family Byte: READ_ALGORITHM_CONFIG (0x51) and Index
+  // Byte: READ_ALGORITHM_INDEX_BYTE - AGC
+  enum READ_AGC_ALGO_WRITE_BYTE
+  {
+    READ_AGC_PERC_ID = 0x00,
+    READ_AGC_STEP_SIZE_ID,
+    READ_AGC_SENSITIVITY_ID,
+    READ_AGC_NUM_SAMPLES_ID,
+    READ_MAX_FAST_COEF_ID = 0x0B
+  };
+
+  // Index Byte under the Family Byte: ENABLE_ALGORITHM (0x52).
+  enum ALGORITHM_MODE_ENABLE_INDEX_BYTE
+  {
+    ENABLE_AGC_ALGO = 0x00,
+    ENABLE_WHRM_ALGO = 0x02
+  };
+
+  class sparkfun_MAX32664
+  {
+  public:
+    sparkfun_MAX32664(PinName sda, PinName scl) : i2c_hb(sda, scl)
+    {
+    }
+
+    uint8_t begin(void);
+
+    uint8_t configBpm(uint8_t);
+
+    struct bioData readBpm(void);
+
+    uint8_t setOutputMode(uint8_t);
+
+    uint8_t setFifoThreshold(uint8_t);
+
+    uint8_t agcAlgoControl(uint8_t);
+
+    uint8_t max30101Control(uint8_t);
+
+    uint8_t maximFastAlgoControl(uint8_t);
+
+    uint8_t readAlgoSamples(void);
+
+    uint8_t readSensorHubStatus(void);
+
+    uint8_t numSamplesOutFifo(void);
+
+    void heartbeat__initialize_application_mode();
+
+    uint8_t enableWrite(uint8_t, uint8_t, uint8_t);
+
+    uint8_t writeByte_1(uint8_t, uint8_t, uint8_t);
+
+    uint8_t readByte_1(uint8_t, uint8_t);
+
+    uint8_t readByte_2(uint8_t, uint8_t, uint8_t);
+
+    uint8_t readFillArray(uint8_t, uint8_t, uint8_t, uint8_t *);
+
+    void readSensorVersion();
+
+    void hb_start(void);
+
+    void hb_thread_task(void);
+
+  private:
+    I2C i2c_hb;
+
+    Thread thread_;
+
+    uint8_t _address;
+
+    uint8_t _userSelectedMode;
+
+    uint8_t _sampleRate;
+
+    uint8_t bpmArr[MAXFAST_ARRAY_SIZE];
+
+    uint8_t bpmArrTwo[MAXFAST_ARRAY_SIZE + MAXFAST_EXTENDED_DATA];
+    const uint8_t BIO_ADDRESS = 0xAA;
+    const uint8_t INCORR_PARAM = 0xEE;
+    const uint32_t ENABLE_CMD_DELAY_MS = 45; // Milliseconds
+    const uint32_t CMD_DELAY_MS = 6;         // Milliseconds
+    const uint8_t SET_FORMAT = 0x00;
+    const uint8_t ENABLE = 0x01;
+    const uint8_t WRITE_SET_THRESHOLD = 0x01;
+  };
+
+  const bioData &get_hb_data(void);
```

Comment:
Put the getter inside the class right? Why is this outside?
As we discussed when there are multiple instances of this class how will this function behave?

Reply:
Will resolve in next commit.

Reply:
@ganeshram93 This hasnt been resolved correctly.
Make this inside the class?